### PR TITLE
Weatherstation

### PR DESCRIPTION
**Pull request by rsouza01**
* Interface WeatherStation updated.
* Abstract class WeatherStation updated.
* Class FakeWeatherStation created, fully implementing the WeatherStation interface, along with unit conversions for non default units.

### CODE REVIEW

**Comment by wschoenell** [comment]
@rsouza01 nice job!
I would just change one thing... I would move the unit conversion functions to a general function on instruments/weatherstation.py to be inherited by all the weather station drivers.
Something like:

```python
def _convertUnits(value, unit_in, unit_out):
	if unit_in == unit_out: return value
	(do the conversion here...)
	return value converted to unit_out
```
	
sample call:

```
_convertUnits(10, Units.CELSIUS, Units.KELVIN) = 283
```

**Comment by phsilva** [comment]
And maybe try to use astropy units for the unit conversion (https://astropy.readthedocs.org/en/latest/units/), but not sure if might be overkill at this moment, just don't like the smell of constants rolling around the code.

**Comment by wschoenell** [comment]
:+1: for the `astropy.units` use. :-)

**Comment by tribeiro** [comment]
:+1: `astropy.units`

**Comment by wschoenell** [comment]
Something that I see that maybe could be a problem on this interface is the lack of date/time information. Imagine that one have a weather station which sends data only every 10 minutes. Would him want to save the information that the relative humidity was actually from 10 minutes ago? How? `temperature()` and the other methods returning `value, datetime_measurement`?

**Comment by phsilva** [comment]
returning (t, value) looks logical to me, a namedtuple if possible, tuples are just confuse, you never know what value[0] means vs value.t or value.temperature.

**Comment by rsouza01** [comment]
System of units changed from proprietary enum to astropy.units.

**Comment by rsouza01** [comment]
Named tuples implemented as requested.

**Comment by wschoenell** [comment]
* Why do you need all these configuration keys?:
https://github.com/rsouza01/chimera/blob/aaa8869136160f4ffc0fe92f41e7dbfbdb5dc5d3/src/chimera/interfaces/weatherstation.py#L123-L135

* Can you move the configuration to the beginning of the file? And document it like in?:
https://github.com/astroufsc/chimera/blob/3056296fc951d09a3765b06499a9c8a001363129/src/chimera/interfaces/autoguider.py#L35-L44

* I miss the `model` keyword. Without it, breaks your code here:
https://github.com/rsouza01/chimera/blob/aaa8869136160f4ffc0fe92f41e7dbfbdb5dc5d3/src/chimera/instruments/weatherstation.py#L68

* After you changed to the namedtuples you did not changed these lines:
https://github.com/rsouza01/chimera/blob/aaa8869136160f4ffc0fe92f41e7dbfbdb5dc5d3/src/chimera/instruments/weatherstation.py#L69-L75

They make the header look like this:
screen shot 2015-08-03 at 16 14 47

**Comment by rsouza01** [comment]
Fixes and documentation implemented.

File: src/chimera/interfaces/weatherstation.py [hunk 1]
```diff
@@ -21,100 +21,125 @@
...
+        :return: the temperature.
+        """
         pass
 
-    @event
-    def temperatureChange(self, temperature, unit, delta):
+    def wind_speed(self, unit_out=units.kilometer/units.hour):
+        """
+        Returns the wind speed in the chosen unit (Default: Meters per second).
+        :param unit:  Unit in which the instrument should return the wind speed.
```
Comment (wschoenell) [inline src/chimera/interfaces/weatherstation.py]: You say in the doc that it is m/s, but on def is km/h. I think that the default should be m/s.

File: src/chimera/interfaces/weatherstation.py [hunk 2]
```diff
@@ -21,100 +21,125 @@
...
         pass
 
-    @event
-    def pressureChange(self, pressure, unit, delta):
+    def pressure(self, unit_out=units.cds.mmHg):
+        """
+        Returns the pressure in the chosen unit (Default: mmHg).
+        :param unit:  Unit in which the instrument should return the pressure.
+        :return: the pressure.
+        """
```
Comment (wschoenell) [inline src/chimera/interfaces/weatherstation.py]: I think that here the default unit should be Pascal (to be SI consistent).

**Comment by wschoenell** [comment]
@rsouza01,

I think that if you do the minor changes that I just commented on the code and add the correct the units to the metadata output (the `[XXX]` below), this PR is okay to merge.

	METMODEL= 'unknown '           / Weather station Model                          
	METRH   = '94.6410161514'      / [%] Weather station relative humidity          
	METTEMP = '-6.65063509461'     / [degC] Weather station temperature             
	METWINDS= '10      '           / [m/s] Weather station wind speed               
	WINDDIR = '90.0    '           / [deg] Weather station wind direction           
	METDEW  = '10      '           / [degC] Weather station dew point               
	METPRES = '1140.0  '           / [hPa] Weather station air pressure             
	METRAIN = '0       '           / Weather station rain indicator                 

I think that it will be something like `temperature().unit.name` inside that bracket.

**Comment by rsouza01** [comment]
All minor changes implemented.

File: src/chimera/interfaces/weatherstation.py [hunk 3]
```diff
@@ -21,100 +21,125 @@
...
         pass
 
-    @event
-    def rainChange(self, rain, unit, delta):
-        pass
+    def rain(self, unit_out=units.liter/units.hour):
+        """
+        Returns the precipitation rate in the chosen unit (Default: mm/H).
```
Comment (wschoenell) [inline src/chimera/interfaces/weatherstation.py]: Here is still wrong the doc in respect to the def. Precipitation rate unit can be mm/hour or inches/hour and not volume/hour.

http://web.ead.anl.gov/resrad/datacoll/water.htm

Comment (ankanaan) [inline src/chimera/interfaces/weatherstation.py]: not sure in what direction the sentence goes.

just to be clear:

mm/hour or inches/hour is right

volume/time is wrong

On Tue, Aug 4, 2015 at 12:30 PM, William Schoenell <notifications@github.com

**Comment by rsouza01** [comment]
Fixes related to the rain indicator implemented.

**Comment by tribeiro** [comment]
How is this going? Last update was 6 days ago. Can we merge? :grin:

**Comment by wschoenell** [comment]
For me it is good to go. If no one manifests against, I will merge next monday.

Tested on LNA with the fake driver:

    METMODEL= 'unknown '           / Weather station Model                          
    METRH   = '31.7157287525'      / [%] Weather station relative humidity          
    METTEMP = '32.6776695297'      / [degC] Weather station temperature             
    METWINDS= '10      '           / [m/s] Weather station wind speed               
    WINDDIR = '307.279220614'      / [deg] Weather station wind direction           
    METDEW  = '10      '           / [degC] Weather station dew point               
    METPRES = '1140.0  '           / [hPa] Weather station air pressure             
    METRAIN = '0       '           / Weather station rain indicator

**Pull request merged**